### PR TITLE
.swiftversion is created

### PR DESCRIPTION
**Pull request by erkekin**
- swift version file added to prevent some skipped rules
- swiftformat ran without --lint applied

### Short description 📝

Noticed a CI warning recently

```bash
Run ./fourier format swift
Running SwiftFormat...
(lint mode - no files will be changed.)
Reading config file at /Users/runner/work/tuist/tuist/.swiftformat
warning: No Swift version was specified, so some formatting features were disabled. Specify the version of Swift you are using with the --swiftversion option, or by adding a .swift-version file to your project.
```

So I thought we could add `.swift-version` file to enable all rules. Then I ran the command without `--lint` and committed the changes accordingly.

### How to test the changes locally 🧐

When I tried to commit my changes I witnessed this messages as well.

### Checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase.
- [ ] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [ ] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [ ] In case the PR introduces changes that affect users, the documentation has been updated.
- [ ] In case the PR introduces changes that affect how the cache artifact is generated starting from the `TuistGraph.Target`, the `Constants.cacheVersion` has been updated.

### CODE REVIEW

File: .swift-version [hunk 1]
```diff
@@ -0,0 +1 @@
+5.5.1
```
Comment (danyf90) [inline .swift-version:1]: Given we are releasing with Xcode 12.X I don't think we can use 5.5.1 here.
But maybe now that Xcode 13.2 is out we should be able to use it for release as it should solve the compatibility issues we have found.

cc @fortmarek @luispadron

Comment (fortmarek) [reply .swift-version:1]: Isn't this version used only for linting which is not run during release? I think if CI passes, we should be fine. However, it is worth a try whether we can start releasing with 13.2 as we could use Swift 5.5 features such as synthesization of Codable conformance for enums with associated values.

File: projects/fourier/lib/fourier/services/format/swift.rb [hunk 2]
```diff
@@ -12,9 +12,9 @@ def initialize(fix:)
 
         def call
           Dir.chdir(Constants::ROOT_DIRECTORY) do
-            arguments = [vendor_path("swiftformat/swiftformat"), "."]
+            arguments = [vendor_path("swiftformat/swiftformat"), ".", "--swiftversion", "5.5.1"]
             unless fix
-              arguments << "--lint"
```
Comment (ezraberch) [inline projects/fourier/lib/fourier/services/format/swift.rb:17]: Why are you removing this flag?

Comment (erkekin) [reply projects/fourier/lib/fourier/services/format/swift.rb:17]: it was a mistake, thank you!

**Comment by fortmarek** [comment]
@erkekin you'll have to update the checks to use Xcode 13.2 instead 13.0 as the Swift version and Xcode versions currently don't match

File: .swift-version (hunk 1 above)
Comment (danyf90) [reply .swift-version:1]: If swiftformat relies on this for some checks, I imagine it means such API is available only on newer iOS versions 🤷‍♂️

Comment (fortmarek) [reply .swift-version:1]: Not sure how iOS is related here? But it's true we could possibly run into a scenario where `swiftformat` autofixes code in a way that it becomes uncompilable with Xcode 12.X - but that should be caught by one of our checks.

Comment (danyf90) [reply .swift-version:1]: Yeah I meant swift version sorry :)
I think we should put here the minimum version we support Tuist on. If we bump the release to use Xcode 13.2, then we can bump this to 5.5.2 👍

**Review by danyf90 — APPROVED** [review]

**Comment by erkekin** [comment]
> @erkekin you'll have to update the checks to use Xcode 13.2 instead 13.0 as the Swift version and Xcode versions currently don't match

I did it but interestingly it keeps failing 🤔 any ideas why?

**Comment by ezraberch** [comment]
> I did it but interestingly it keeps failing 🤔 any ideas why?

Try changing the target branch from `main` to `xcode13.2`.

**Comment by pepicrft** [comment]
@erkekin thanks a lot for doing this. Once the conflicts are resolved we can go ahead and merge.

**Comment by erkekin** [comment]
> @erkekin thanks a lot for doing this. Once the conflicts are resolved we can go ahead and merge.

You're welcome, as of now this pr could be merged in.

**Review by ezraberch — APPROVED** [review]

**Pull request merged**